onnxruntime/test/testdata/custom_op_local_function/custom_gemm.cc

```diff
@@ -0,0 +1,792 @@
+  if (compute_type == "CUBLAS_COMPUTE_16F") {
+    computeType_ = ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16;
+  } else if (compute_type == "CUBLAS_COMPUTE_32F") {
+    computeType_ = ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT;
+  } else if (compute_type == "CUBLAS_COMPUTE_32F_FAST_16F") {
+    computeType_ = ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT16;
+  } else if (compute_type == "CUBLAS_COMPUTE_32F_FAST_TF32") {
+    computeType_ = ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT;
+  } else {
+    ORT_CXX_API_THROW("Wrong computeType", OrtErrorCode::ORT_RUNTIME_EXCEPTION);
+  }
+
+  std::string activation =
+      KernelInfoGetOptionalAttributeString(api, info, "activation", "DEFUALT");
+  if (activation == "DEFUALT") {
+    epilogue_ = EpiloqueGemmKernel::Default;
+  } else if (activation == "RELU") {
+    epilogue_ = EpiloqueGemmKernel::Relu;
+  } else if (activation == "GELU") {
+    epilogue_ = EpiloqueGemmKernel::Gelu;
+  } else {
+    ORT_CXX_API_THROW("Wrong activation", OrtErrorCode::ORT_RUNTIME_EXCEPTION);
+  }
+
+  ThrowOnError(api, api.KernelInfo_GetInputCount(info, &n_inputs_));
+  if (n_inputs_ > 5) {
+    std::string name = KernelInfoGetInputName(api, info, 5);
+    has_scale_Y_ = !name.empty();
+  } else {
+    has_scale_Y_ = false;
+  }
+}
+
+void CustomGemmKernel::set(const std::vector<int64_t>& shape_A,
+                           const std::vector<int64_t>& shape_B, int& M, int& N,
+                           int& K, int& lda, int& ldb, int& ldd,
+                           int row_major) const {
+  constexpr int ir = 0;
+  constexpr int ic = 1 - ir;
+  if (transA_ && !transB_) {  // TN
+    M = static_cast<int>(shape_A[ic]);
+    N = static_cast<int>(shape_B[ic]);
+    K = static_cast<int>(shape_A[ir]);
+    lda = static_cast<int>(shape_A[row_major ? ic : ir]);
+    ldb = static_cast<int>(shape_B[row_major ? ic : ir]);
+    ldd = static_cast<int>(shape_B[row_major ? ic : ir]);
+  } else if (!transA_ && !transB_) {  // NN
+    M = static_cast<int>(shape_A[ir]);
+    N = static_cast<int>(shape_B[ic]);
+    K = static_cast<int>(shape_A[ic]);
+    lda = static_cast<int>(shape_A[row_major ? ic : ir]);
+    ldb = static_cast<int>(shape_B[row_major ? ic : ir]);
+    ldd = static_cast<int>(shape_B[row_major ? ic : ir]);
+  } else if (!transA_ && transB_) {  // NT
+    M = static_cast<int>(shape_A[ir]);
+    N = static_cast<int>(shape_B[ir]);
+    K = static_cast<int>(shape_A[ic]);
+    lda = static_cast<int>(shape_A[row_major ? ic : ir]);
+    ldb = static_cast<int>(shape_B[row_major ? ic : ir]);
+    ldd = static_cast<int>(shape_B[row_major ? ir : ic]);
+  } else {  // TT
+    M = static_cast<int>(shape_A[ic]);
+    N = static_cast<int>(shape_B[ir]);
+    K = static_cast<int>(shape_A[ir]);
+    lda = static_cast<int>(shape_A[row_major ? ic : ir]);
+    ldb = static_cast<int>(shape_B[row_major ? ic : ir]);
+    ldd = static_cast<int>(shape_B[row_major ? ir : ic]);
+  }
+}
+
+void check_device(const Ort::ConstValue& input, const char* /*name*/) {
```

Comment:
nit: Function names should be PascalCase https://google.github.io/styleguide/cppguide.html#Function_Names

Reply:
removed